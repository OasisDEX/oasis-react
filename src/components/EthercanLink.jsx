import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import styles from './EthercanLink.scss';


const propTypes = PropTypes && {
  label: PropTypes.string,
  networkName: PropTypes.string
};
const defaultProps = {
  networkName: 'kovan'
};


class EthercanLink extends PureComponent {

  render() {
    const { txHash, label, networkName } = this.props;
    const address = `https://${networkName}.etherscan.io/tx/${txHash}`;
    return (
      <span >
        <a style={{ textOverflow: 'elipsis',maxWidth: '80%' }} href={address}>{label||txHash}</a>
      </span>
    );
  }
}

EthercanLink.displayName = 'EthercanLink';
EthercanLink.propTypes = propTypes;
EthercanLink.defaultProps = defaultProps;
export default EthercanLink;
