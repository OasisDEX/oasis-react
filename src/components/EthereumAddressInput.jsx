import React, { PureComponent } from "react";
import { PropTypes } from "prop-types";
import CSSModules from "react-css-modules";
import styles from "./EthereumAddressInput.scss";
// import ImmutablePropTypes from 'react-immutable-proptypes';
const propTypes = PropTypes && {};
const defaultProps = {};

const inputStyle = { textAlign: "right", paddingRight: "30px" };

class EthereumAddressInput extends PureComponent {
  render() {
    return (
      <input
        {...this.props.input}
        disabled={this.props.disabled}
        style={inputStyle}
        autoComplete="off"
        required={this.props.required}
        placeholder={"0x"}
        type="text"
      />
    );
  }
}

EthereumAddressInput.displayName = "EthereumAddressInput";
EthereumAddressInput.propTypes = propTypes;
EthereumAddressInput.defaultProps = defaultProps;
export default CSSModules(EthereumAddressInput, styles);
