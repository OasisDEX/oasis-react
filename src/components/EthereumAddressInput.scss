.EthereumAddressInput {
  input {
    padding-right: 20px !important;
  }
}