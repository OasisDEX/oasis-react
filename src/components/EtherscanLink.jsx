import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

export const ETHERSCAN_LINK_TYPE_ADDRESS = 'ETHERSCAN_LINK_TYPE_ADDRESS';
export const ETHERSCAN_LINK_TYPE_TRANSACTION = 'ETHERSCAN_LINK_TYPE_TRANSACTION';

const propTypes = PropTypes && {
  label: PropTypes.string,
  networkName: PropTypes.string,
  txHash: PropTypes.string,
  address: PropTypes.string,
  type: PropTypes.oneOf(
    [
      ETHERSCAN_LINK_TYPE_ADDRESS,
      ETHERSCAN_LINK_TYPE_TRANSACTION
    ]
  )
};
const defaultProps = {
  networkName: 'kovan'
};




const linkStyle = { maxWidth:'100px', textOverflow: 'ellipsis', display: 'inline-block', overflow: 'hidden' };

class EtherscanLink extends PureComponent {
  render() {
    const { txHash, label, networkName, address } = this.props;
    switch (this.props.type) {
      case ETHERSCAN_LINK_TYPE_TRANSACTION: {
        const url = `https://${networkName}.etherscan.io/tx/${txHash}`;
        return (
          <a style={linkStyle} href={url}>
            <span>{label||txHash}</span>
          </a>
        );
      }
      case ETHERSCAN_LINK_TYPE_ADDRESS: {
        const url = `https://${networkName}.etherscan.io/address/${address}`;
        return (
          <a style={linkStyle} href={url}>
            <span>{label||address}</span>
          </a>
        );
      }
    }
  }
}

EtherscanLink.displayName = 'EthercanLink';
EtherscanLink.propTypes = propTypes;
EtherscanLink.defaultProps = defaultProps;
export default EtherscanLink;
