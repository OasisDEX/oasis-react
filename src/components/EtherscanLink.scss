@import "../styles/modules/breakpoints";
@import "../styles/modules/typography";

.link {
  @include ellipsis(150px);
  @media (max-width: $xxs) {
    @include ellipsis(70px);
  }
  font-weight: 600;
}
