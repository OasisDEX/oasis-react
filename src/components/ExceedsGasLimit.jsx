import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

const propTypes = PropTypes && {
};
const defaultProps = {};


class ExceedsGasLimit extends PureComponent {
  render() {
    return (
      <div>
        <div>
          Your order exceed gas limit of <b>4.3 Million</b>.
        </div>
        <div>
          To get a working order use the <b>Buy Max Button</b>
        </div>
      </div>
    );
  }
}

ExceedsGasLimit.displayName = 'ExceedsGasLimit';
ExceedsGasLimit.propTypes = propTypes;
ExceedsGasLimit.defaultProps = defaultProps;
export default ExceedsGasLimit;
