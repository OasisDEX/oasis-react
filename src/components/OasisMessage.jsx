import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';

import styles from './OasisMessage.scss';
import CSSModules from 'react-css-modules';

export const MSGTYPE_INFO = 'MSGTYPE_INFO';
export const MSGTYPE_WARNING = 'MSGTYPE_WARNING';

const BOARD_TYPE = {
  [MSGTYPE_INFO]: 'MessageBoard--Info',
  [MSGTYPE_WARNING]: 'MessageBoard--Warning',
};

const propTypes = PropTypes && {
  children: PropTypes.node.isRequired,
  heading: PropTypes.string.isRequired,
  type: PropTypes.oneOf([MSGTYPE_INFO, MSGTYPE_WARNING]),
  dismissible: PropTypes.bool.isRequired,
  onDismiss: PropTypes.func
};
const defaultProps = {
  dismissible: false,
};

export class OasisMessage extends PureComponent {
  constructor(props) {
    super(props);
    this.onDismiss = this.onDismiss.bind(this);
  }

  onDismiss() {
    const { type } = this.props;
    this.props.onDismiss(type);
  }

  render() {
    const { dismissible, hidden } = this.props;
    return (
      <div hidden={hidden} styleName={`MessageBoard ${BOARD_TYPE[this.props.type]}`}>
        <h3 styleName="Heading">
          {this.props.heading}
          {dismissible && <span onClick={this.onDismiss} styleName="Close"/>}
        </h3>
        <div>
          {this.props.children}
        </div>
      </div>
    );
  }
}

OasisMessage.displayName = 'OasisMessage';
OasisMessage.propTypes = propTypes;
OasisMessage.defaultProps = defaultProps;
export default CSSModules(OasisMessage, styles, { allowMultiple: true });
