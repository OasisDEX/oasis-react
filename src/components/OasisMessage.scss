@import "../styles/modules/labels";
@import "../styles/modules/shapes";

.MessageBoard {
  position: relative;
  height: 100%;
  background: rgba(215, 215, 215, 0.05);
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.15);
  border-radius: 3px;
  padding: 8px 24px 8px 20px;

  & .Heading {
    position: relative;
    padding: 10px 0;

    @include text(13px, 700);
    text-align: center;
    letter-spacing: 0.54px;
    text-transform: uppercase;
  }

  &--Info {
    .Heading {
      color: rgba(26, 188, 156, 0.68);
    }
  }
  &--Warning {
    .Heading {
      color: rgba(240, 115, 113, 0.68);
    }
  }

  ul {
    padding-left: 0;
    list-style-type: none;

    & li {
      display: flex;
      align-items: center;
      margin-bottom: 8px;
      letter-spacing: 0.58px;
    }
  }

}

.Circle {
  @include circle(14px);
  margin-right: 24px;
  border-color: darkgray;

  .Number {
    color: darkgray;
    position: relative;
    left: 0.5px;
  }

  .FilledCircle {
    @include circle(8px);
    background-color: rgba(240, 115, 113, 0.68);;
  }

  &--Red {
    border-color: rgba(240, 115, 113, 0.68);
  }
}

.Text {
  display: inline-block;
  @include text(12px, 700);
  color: #4F5362;
  opacity: 0.68;
  width: calc(100% - 40px);;
}

.Close {
  @extend .Cross;

  position: absolute;
  height: 10px;
  width: 10px;
  right: 10px;

  cursor: pointer;
}