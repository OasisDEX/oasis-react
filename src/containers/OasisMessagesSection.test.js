/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import Immutable from 'immutable';

import {
  OasisMessagesSectionWrapper,
  mapStateToProps,
  mapDispatchToProps,
} from './OasisMessagesSection';
import { shallow } from 'enzyme';
import { MSGTYPE_INFO, MSGTYPE_WARNING } from '../components/OasisMessage';

describe('(Container) OasisMessagesSection', () => {
  const state = Immutable.fromJS({
    session: {
      persist: {
        messages: {
          [MSGTYPE_WARNING]: {},
          [MSGTYPE_INFO]: {
            dismissed: false
          }
        }
      }
    }
  });
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });

  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <OasisMessagesSectionWrapper {...props}/>,
    );
    expect(wrapper).toMatchSnapshot();
  });

});
