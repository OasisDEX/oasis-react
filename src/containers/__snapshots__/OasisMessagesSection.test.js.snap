// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) OasisMessagesSection should render 1`] = `
<OasisMessage
  dismissible={false}
  heading="Loading Messages..."
  type="MSGTYPE_INFO"
>
  <span
    hidden={true}
  />
</OasisMessage>
`;

exports[`(Container) OasisMessagesSection will receive right actions 1`] = `
Object {
  "actions": Object {},
}
`;

exports[`(Container) OasisMessagesSection will receive right props 1`] = `
Object {
  "isSessionInitialized": undefined,
  "messages": Object {
    "MSGTYPE_INFO": Object {
      "dismissed": false,
    },
    "MSGTYPE_WARNING": Object {},
  },
}
`;
